src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        final int countOfNames = names.length;
+        final int[] employeesSalaryCount = new int[countOfNames];
+        final LocalDate dateFromFormatted = LocalDate.parse(dateFrom, formatter);
+        final LocalDate dateToFormatted = LocalDate.parse(dateTo, formatter);
+        StringBuilder resultForPeriod = new StringBuilder();
+        resultForPeriod.append("Report for period " + dateFrom + " - " + dateTo);
+        for (String datum : data) {
+            LocalDate currentDate = LocalDate.parse(
+                    datum.substring(0, datum.indexOf(" ")), formatter);
+            if (dateCheck(dateFromFormatted, dateToFormatted, currentDate)) {
+                for (int j = 0; j < names.length; j++) {
+                    if (datum.contains(names[j])) {
+                        employeesSalaryCount[j] += dailySalaryCalculation(datum);
+                    }
+                }
+            }
+        }
+        for (int i = 0; i < countOfNames; i++) {
+            resultForPeriod.append(
+                    System.lineSeparator() + names[i] + " - " + employeesSalaryCount[i]);
+        }
+        return resultForPeriod.toString();
```

Comment:
![image](https://user-images.githubusercontent.com/81781479/195994689-abb8e0b6-b3d3-4c03-a6ee-522a7777a9a5.png)

lets iterate over names , and then over data
for (String name: names) {
    for(String record: data) {
        ....
    }
    reportBuilder.append(...)
}
this way you will avoid the third loop

Reply:
That was easier than I thought. Thank you!